opening_hours/models.py

```diff
@@ -0,0 +1,103 @@
+from datetime import date, datetime, time, timedelta
+
+from django.db import models
+from django.db.models import Case, F, Q, QuerySet, Value, When
+from django.utils.timezone import get_default_timezone
+from django.utils.translation import gettext_lazy as _
+
+DEFAULT_TIMEZONE = get_default_timezone()
+
+
+class OriginHaukiResource(models.Model):
+    # Resource id in Hauki API
+    id = models.IntegerField(unique=True, primary_key=True)
+    # Hauki API hash for opening hours, which is used to determine if the opening hours have changed
+    opening_hours_hash = models.CharField(max_length=64, blank=True)
+    # Latest date fetched from Hauki opening hours API
+    latest_fetched_date = models.DateField(blank=True, null=True)
+
+    def __str__(self) -> str:
+        return str(self.id)
+
+    def is_reservable(self, start_datetime: datetime, end_datetime: datetime) -> bool:
+        return self.reservable_time_spans.filter_period(start=start_datetime, end=end_datetime).exists()
+
+
+class ReservableTimeSpanQuerySet(models.QuerySet):
```

Comment:
Would put this QuerySet class to another file, e.g. `/opening_hours/querysets.py` so that the models.py is dedicated to only model classes.